src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -0,0 +1,390 @@
+    public NBTTagCompound getUpdateTag()
+    {
+        return writeToNBT(new NBTTagCompound());
+    }
+
+    @Override
+    public void update()
+    {
+        super.update();
+
+        if(getColony() != null)
+        {
+            Map<BlockPos, AbstractBuilding> buildingMap = getColony().getBuildings();
+
+            if (buildingMap.size() < this.index)
+            {
+                this.index = 1;
+            }
+
+            int i = 1;
+            for (Map.Entry<BlockPos, AbstractBuilding> buildingEntry : buildingMap.entrySet())
+            {
+                if (i == index)
+                {
+                    if(buildingEntry.getValue() instanceof AbstractBuildingWorker
+                            && !list.contains(buildingEntry.getValue())
+                            && ((AbstractBuildingWorker) buildingEntry.getValue()).needsAnything())
+                    {
+                        checkInWareHouse((AbstractBuildingWorker) buildingEntry.getValue());
+                    }
+                    this.index++;
+                }
+                i++;
+            }
+        }
+    }
+
+    /**
+     * Get the first task in the list.
+     * @return the building which needs a delivery.
+     */
+    @Nullable
+    public AbstractBuilding getTask()
+    {
+        if(list.isEmpty())
+        {
+            return null;
+        }
+        return list.remove(0);
+    }
+
+    /**
+     * Check if the required items by the building are in the wareHouse.
+     * @param buildingEntry the building requesting.
+     * @return true if has something in warehouse to deliver.
+     */
+    public boolean checkInWareHouse(final AbstractBuilding buildingEntry)
+    {
+        if(buildingEntry.areItemsNeeded())
+        {
+            for(ItemStack stack : buildingEntry.getNeededItems())
+            {
+                if(isInHut(stack))
+                {
+                    buildingEntry.setOnGoingDelivery(true);
+                    list.add(buildingEntry);
+                    return true;
+                }
+            }
+        }
+
+        String tool = buildingEntry.getRequiredTool();
+        if(!tool.isEmpty() && isToolInHut(tool))
+        {
+            buildingEntry.setOnGoingDelivery(true);
+            list.add(buildingEntry);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Check all chests in the worker hut for a required item.
+     * @param is the type of item requested (amount is ignored)
+     * @return true if a stack of that type was found
+     */
+    public boolean isInHut(@Nullable final ItemStack is)
+    {
+        @Nullable final AbstractBuilding building = getBuilding();
+
+        boolean hasItem;
+        if(building != null)
+        {
+            hasItem = isInTileEntity(building.getTileEntity(), is);
+
+            if(hasItem)
+            {
+                return true;
+            }
+
+            for(BlockPos pos : building.getAdditionalCountainers())
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'pos' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)